Scripts/ThrusterAsset.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+[CreateAssetMenu]
+public class ThrusterAsset : ScriptableObject
+{
+    public AnimationCurve curve = AnimationCurve.Linear(0, 0, 1, 1);
+    [HideInInspector]
+    public AnimationCurve inversedCurve = AnimationCurve.Linear(0,0,1,1);
+ 
+    public static implicit operator AnimationCurve(ThrusterAsset me)
+    {
+        me.inversedCurve = inverseCurve(me.curve);
+        return me.curve;
+    }
+    public static implicit operator ThrusterAsset(AnimationCurve curve)
+    {
+        ThrusterAsset asset = ScriptableObject.CreateInstance<ThrusterAsset>();
+        asset.curve = curve;
+        asset.inversedCurve = inverseCurve(asset.curve);
+        return asset;
+    }
+
+    public static AnimationCurve inverseCurve(AnimationCurve curve)
```

Comment:
Whenever somebody calles inverseCurve, it is calculated again. This inverse should be set once in the constructor of this class

Reply:
better make this private